packages/visual-editor/src/utils/assemblyUtils.ts

```diff
@@ -65,24 +66,24 @@ export const deserializeDesignComponentNode = ({
     }
   }
 
-  const isDesignComponent = designComponentsRegistry.has(node.definitionId);
+  const isAssembly = assembliesRegistry.has(node.definitionId);
 
   const children: CompositionComponentNode[] = node.children.map((child) =>
-    deserializeDesignComponentNode({
+    deserializeAssemblyNode({
       node: child,
       nodeId: generateRandomId(16),
       parentId: nodeId,
-      designComponentDataSource,
-      designComponentUnboundValues,
+      assemblyDataSource,
+      assemblyUnboundValues,
       componentInstanceProps,
       componentInstanceUnboundValues,
       componentInstanceDataSource,
     })
   );
 
   return {
-    // separate node type identifiers for design components and their blocks, so we can treat them differently in as much as we want
-    type: isDesignComponent ? DESIGN_COMPONENT_NODE_TYPE : DESIGN_COMPONENT_BLOCK_NODE_TYPE,
+    // separate node type identifiers for assemblies and their blocks, so we can treat them differently in as much as we want
+    type: isAssembly ? ASSEMBLY_NODE_TYPE : ASSEMBLY_BLOCK_NODE_TYPE,
```

Comment:
If using an older version of the SDK here the ASSEMBLY_NODE_TYPE will be undefined, which will cause issues later on, correct? Does this mean this is a breaking change and can only work on latest SDK?